syntax = "proto3";


import "google/protobuf/timestamp.proto";
import "google/protobuf/wrappers.proto";
import "google/type/date.proto";
import "google/type/timeofday.proto";


package protarrow_protos;


enum TestEnum {
  UNKNOWN_TEST_ENUM = 0;
  TEST_ENUM_1 = 1;
  TEST_ENUM_2 = 2;
}

message TestMessage {
  double double_value = 1;
  float float_value = 2;
  int32 int32_value = 3;
  int64 int64_value = 4;
  uint32 uint32_value = 5;
  uint64 uint64_value = 6;
  sint32 sint32_value = 7;
  sint64 sint64_value = 8;
  fixed32 fixed32_value = 9;
  fixed64 fixed64_value = 10;
  sfixed32 sfixed32_value = 11;
  sfixed64 sfixed64_value = 12;
  bool bool_value = 13;
  string string_value = 14;
  bytes bytes_value = 15;
  TestEnum enum_value = 16;
  google.protobuf.Timestamp timestamp = 17;
  google.type.Date date = 18;
  google.type.TimeOfDay time_of_day = 19;
  google.protobuf.DoubleValue wrapped_double = 20;
  google.protobuf.FloatValue wrapped_float = 21;
  google.protobuf.Int32Value wrapped_int32 = 22;
  google.protobuf.Int64Value wrapped_int64 = 23;
  google.protobuf.UInt32Value wrapped_uint32 = 24;
  google.protobuf.UInt64Value wrapped_uint64 = 25;
  google.protobuf.BoolValue wrapped_bool = 26;
  google.protobuf.StringValue wrapped_string = 27;
  google.protobuf.BytesValue wrapped_bytes = 28;

  repeated double double_values = 101;
  repeated float float_values = 102;
  repeated int32 int32_values = 103;
  repeated int64 int64_values = 104;
  repeated uint32 uint32_values = 105;
  repeated uint64 uint64_values = 106;
  repeated sint32 sint32_values = 107;
  repeated sint64 sint64_values = 108;
  repeated fixed32 fixed32_values = 109;
  repeated fixed64 fixed64_values = 110;
  repeated sfixed32 sfixed32_values = 111;
  repeated sfixed64 sfixed64_values = 112;
  repeated bool bool_values = 113;
  repeated string string_values = 114;
  repeated bytes bytes_values = 115;
  repeated TestEnum enum_values = 116;
  repeated google.protobuf.Timestamp timestamps = 117;
  repeated google.type.Date dates = 118;
  repeated google.type.TimeOfDay time_of_days = 119;

  map<string, double> double_map = 201;
  map<string, float> float_map = 202;
  map<string, int32> int32_map = 203;
  map<string, int64> int64_map = 204;
  map<string, uint32> uint32_map = 205;
  map<string, uint64> uint64_map = 206;
  map<string, sint32> sint32_map = 207;
  map<string, sint64> sint64_map = 208;
  map<string, fixed32> fixed32_map = 209;
  map<string, fixed64> fixed64_map = 210;
  map<string, sfixed32> sfixed32_map = 211;
  map<string, sfixed64> sfixed64_map = 212;
  map<string, bool> bool_map = 213;
  map<string, string> string_map = 214;
  map<string, bytes> bytes_map = 215;
  map<string, TestEnum> enum_map = 216;
  map<string, google.protobuf.Timestamp> timestamp_map = 217;
  map<string, google.type.Date> date_map = 218;
  map<string, google.type.TimeOfDay> time_of_day_map = 219;
}

message NestedTestMessage {
  TestMessage test_message = 1;
  repeated TestMessage test_messages = 2;
  map<string, TestMessage> test_message_map = 3;
}
